### PR TITLE
Add a tool for converting downloaded models

### PR DESCRIPTION
**Pull request by IRDonch**
~Also, move the model downloader and the new tool under `tools/automation` and call them the OMZ automation tools.~

### CODE REVIEW

**Comment by IRDonch** [comment]
This is a draft, because the CI is not ready for the new location of the tools, and there are still a few mentions of the old model downloader path in the models and demos. I just want to get some early feedback.

File: tools/automation/README.md [hunk 1]
```diff
@@ -0,0 +1,212 @@
+Open Model Zoo automation tools
+===============================
+
+This directory contains scripts that automate certain model-related tasks
+based on the included configuration file.
+
+* `downloader.py` (model downloader) downloads model files from the Internet
```
Comment (zmaslova) [inline tools/automation/README.md]: `internet` -> `public sources`?

File: tools/automation/converter.py [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+        help='Print the conversion commands without running them')
+    args = parser.parse_args()
+
+    mo_path = args.mo
+    if mo_path is None:
+        try:
+            mo_path = Path(os.environ['INTEL_OPENVINO_DIR']) / 'deployment_tools/model_optimizer/mo.py'
+        except KeyError:
+            sys.exit('Unable to locate Model Optimizer. '
+                + 'Use --mo or run setupvars.py/setupvars.bat from the OpenVINO toolkit.')
```
Comment (zmaslova) [inline tools/automation/converter.py]: `setupvars.py` -> `setupvars.sh`

**Comment by yury-gorbachev** [comment]
This path is confusing. I understand tools, but automation sounds weird.

**Comment by kchechil** [comment]
@yury-gorbachev what is your proposal about path name ?

**Comment by zmaslova** [comment]
@IRDonch python code-LGTM

File: tools/automation/converter.py (hunk 2 above)
Comment (IRDonch) [reply tools/automation/converter.py]: Fixed, thanks.

File: tools/automation/README.md (hunk 1 above)
Comment (IRDonch) [reply tools/automation/README.md]: Well... it wouldn't be incorrect, but is it better?

**Comment by yury-gorbachev** [comment]
> @yury-gorbachev what is your proposal about path name ?

How about tools/downloader? This is a bit more intuitive.

**Comment by IRDonch** [comment]
> How about tools/downloader? This is a bit more intuitive.

But incorrect, because it's not just the downloader anymore.

**Comment by yury-gorbachev** [comment]
> > How about tools/downloader? This is a bit more intuitive.
> 
> But incorrect, because it's not just the downloader anymore.

This does not matter much. What matters is that folder named automation is last thing you are looking for when trying to find scripts that download models.
BTW, separation into 2 scripts is also questionable to me, but I understand that there are nuclear science - like issues to be solved to avoid this.

File: tools/automation/README.md (hunk 1 above)
Comment (IRDonch) [reply tools/automation/README.md]: I ended up replacing it with "online sources", which IMO combines the best of both wordings.

**Comment by IRDonch** [comment]
> This does not matter much.

I don't agree, but I also don't want to argue about this, so I renamed it.

**Comment by eizamaliev** [comment]
It's good support PyTorch models as well, because now MO cannot convert them, only in ONNX format. So, I propose to add conversion [script](https://github.com/opencv/openvino_training_extensions/tree/develop/pytorch_toolkit/utils) from Pytorch to ONNX.

**Comment by IRDonch** [comment]
@eizamaliev One feature at a time.

**Comment by IRDonch** [comment]
OK, I'm going to do a slight change of plans here. I can't quite un-draft this, because the documentation needs to be updated with the new path to the tools within the OpenVINO toolkit, and the process of establishing what that path will be is dragging on.

So I'm going to edit this MR to keep just the addition of the model converter (which should be utterly uncontroversial), and leave the change of paths for a later date.

**Comment by kchechil** [comment]
merging as CI failure is purely due to issues with connection to http://dl.yf.io/dilation/models/dilation10_cityscapes.caffemodel
cc @atsyplen to check this link later.

File: tools/automation/converter.py [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+
+if platform.system() == 'Windows':
+    quote_arg = quote_windows
+else:
+    quote_arg = shlex.quote
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('-c', '--config', type=Path, metavar='CONFIG.YML',
+        default=Path(__file__).resolve().parent / 'list_topologies.yml', help='topology configuration file')
```
[the remark this answers is not in the document]
Comment (IRDonch) [reply tools/automation/converter.py]: Yes, naturally.

(Note to self: move the default to `common` at some point...)

File: tools/automation/converter.py [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+            *expanded_mo_args]
+
+        print('========= {}Converting {}'.format('(DRY RUN) ' if args.dry_run else '', top.name))
+
+        print('Conversion command:', ' '.join(map(quote_arg, mo_cmd)))
+
+        if not args.dry_run:
+            print(flush=True)
+
+            if subprocess.run(mo_cmd).returncode != 0:
```
[the remark this answers is not in the document]
Comment (IRDonch) [reply tools/automation/converter.py]: Sure, it should be quite doable.

**Pull request merged**